tools/font-edit/twin-fedit.c

```diff
@@ -492,95 +497,102 @@ static void button(char_t *c, XButtonEvent *bev)
 
 static void play(char_t *c)
 {
-    XEvent ev;
-    char key_string[10];
-
-    XClearArea(dpy, win, 0, 0, 0, 0, True);
-    for (;;) {
-        XNextEvent(dpy, &ev);
-        switch (ev.type) {
-        case KeyPress:
-            if (XLookupString((XKeyEvent *) &ev, key_string, sizeof(key_string),
-                              0, 0) == 1) {
-                switch (key_string[0]) {
-                case 'q':
-                    return;
-                case 'c':
-                    XClearArea(dpy, ev.xkey.window, 0, 0, 0, 0, True);
+    /* Ensures that SDL only focuses on key events*/
+    SDL_StopTextInput();
+
+    SDL_Event event;
+    int quit_state = 0;
```

Comment:
It is confusing to have both `quit_state` and `exit_window`. Think of the way to simplify.

Reply:
Resolved in 7231d62d0106fb56594ddd7785efcc9ab9779b04.

I dropped the `quit_state` and used a for-loop to replace the while-loop, leaving `exit_window` to control the loop flow.